### PR TITLE
Added Loading images from camera using CameraLoader and image display when qt is supported

### PR DESCRIPTION
**Pull request by Mithul**
You can load the images from the camera instead of a folder by specifying a ```-camera_id <camera_id>``` option to the script. The default camera_id for most OS is 0.

The images being processed can now be seen while processing with their captions(as the title of the window) if 'qt' is supported. Using qlua instead of th works.

### CODE REVIEW

**Comment by Mithul** [comment]
fixes #26 .

**Comment by karpathy** [comment]
Thank you for this PR, a few concerns before we merge:

- Could you please exclude changes to `.gitignore`?
- In `eval.lua`, could we do a (safer) lazy loading of the `camera/misc.CameraLoader`? i.e. you could move these parts to right before you instantiate the `CameraLoader`  inside the `if` block.
- Similarly, it seems that the `display_image` function should only be called in case we're actually using the camera loader?
- Spurious `-- break` should be removed
- I don't think the `repeat ... until` logic is backwards compatible? Would this result in an infinite loop?

In short, I would like this somewhat exotic feature to have minimal impact on the script. If the user would like to not use it, it would be nice to have identical behavior as before and with no errors. Almost there!

**Comment by Mithul** [comment]
I have made the changes, the repeat untill will only result in an infinite loop if camera_id is set(which is what is needed I think?). In other cases it would execute only once.

**Comment by SaMnCo** [comment]
That's a great addition. @karpathy when do you think it will be merged? 

Thx

**Comment by karpathy** [comment]
I'm sorry I'm currently at a conference and have become unresponsive on several issues. I'll look at this when I get back early next week and we'll work to merge it in.
